fili-core/src/main/java/com/yahoo/bard/webservice/table/LogicalTableSchema.java

```diff
@@ -45,10 +49,30 @@ public LogicalTableSchema(TableGroup tableGroup, Granularity granularity, Metric
         return Stream.concat(
                 tableGroup.getDimensions().stream()
                 .map(DimensionColumn::new),
-                tableGroup.getApiMetricNames().stream()
-                        .filter(apiMetricName ->  apiMetricName.isValidFor(granularity))
-                        .map(ApiMetricName::getApiName)
-                        .map(name -> new LogicalMetricColumn(name, metricDictionary.get(name)))
+                buildMetricColumns(tableGroup.getApiMetricNames(), granularity, metricDictionary)
         ).collect(Collectors.toCollection(LinkedHashSet::new));
     }
+
+    /**
+     * Buid a stream of metric columns by filtering the table group's collection of ApiMetricNames.
+     * Metric names are resolved and then filtered considering the table granularity and the metric referenced.
+     *
+     * @param apiMetricNames  The names of the apiMetrics being bound and filtered
+     * @param granularity  The grain used to filter those metric names
+     * @param metricDictionary  The dictionary to resolve the logical metric instances from
+     *
+     * @return A stream of metric columns, filtered for compatibility with the grain.
+     */
+    private static Stream<MetricColumn> buildMetricColumns(
```

Comment:
Should this return `Stream<MetricColumn>` or `Stream<LogicalMetricColumn>`? If the former, why the type relaxation?